implementation/src/main/java/io/smallrye/config/ConfigValueConfigSourceInterceptor.java

```diff
@@ -0,0 +1,42 @@
+package io.smallrye.config;
+
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.regex.Pattern;
+
+import javax.annotation.Priority;
+
+import org.jboss.logging.Logger;
+
+@Priority(100)
+public class ConfigValueConfigSourceInterceptor implements ConfigSourceInterceptor {
+    private static final Logger LOG = Logger.getLogger("io.smallrye.config");
+    private static final String EXCLUDE_KEYS_PATTERN_KEY = "smallrye.log.exclude.keys.pattern";
+
+    private AtomicReference<Pattern> pattern = new AtomicReference<>();
+
+    @Override
+    public ConfigValue getValue(final ConfigSourceInterceptorContext context, final String name) {
+        ensureInitPattern(context);
+
+        final ConfigValue configValue = context.proceed(name);
+        if (!pattern.get().matcher(configValue.getName()).matches()) {
+            LOG.infof("The configuration %s with the value %s was loaded from %s:%d",
+                    configValue.getName(),
+                    configValue.getValue(),
+                    configValue.getConfigSourceName(),
+                    configValue.getLineNumber());
+        }
+
+        return configValue;
+    }
+
+    private void ensureInitPattern(final ConfigSourceInterceptorContext context) {
```

Comment:
This is still incorrect.  Other threads can observe a time where the pattern is "".